mongo/index_view.go

```diff
@@ -416,6 +419,15 @@ func (iv IndexView) drop(ctx context.Context, name string, opts ...*options.Drop
 	return res, nil
 }
 
+func (iv IndexView) dropFromKeys(ctx context.Context, keySpecDocument bsoncore.Document, opts ...*options.DropIndexesOptions) (bson.Raw, error) {
+	model := IndexModel{nil, nil}
+	name, err := getOrGenerateIndexName(keySpecDocument, model)
```

Comment:
Using getOrGenerateIndexName doesn't quite accomplish the goal here. That function will take a keySpecDocument and increment it, which would be the name in [default case](https://github.com/mongodb/mongo-go-driver/blob/b10dc345d022fc64ac4d9bc5796c830a7d8c71c5/mongo/index_view.go#L199), but otherwise not. For example this construction would not result in a key being dropped in the proposal:

```go
indexModel := mongo.IndexModel{
	Keys:    bson.M{"username": 1},
	Options: options.Index().SetUnique(true).SetName("myidx"),
}
```

See [this gist](https://gist.github.com/prestonvasquez/70e33be46343ee01a352e2d49a2cec6c) for more details on re-creating the issue.

Can we add a new constructor called NewDropIndexesByKey to the operation layer, similar to [NewDropIndexes](https://github.com/mongodb/mongo-go-driver/blob/b10dc345d022fc64ac4d9bc5796c830a7d8c71c5/x/mongo/driver/operation/drop_indexes.go#L68) that defines the key document that should be used to drop an index? Would it make more since to simply extend the existing NewDropIndexes to accept a key specification document? If so extend the logic here [here](https://github.com/mongodb/mongo-go-driver/blob/b10dc345d022fc64ac4d9bc5796c830a7d8c71c5/x/mongo/driver/operation/drop_indexes.go#L110) to account for creating a wire message that drop an index by key. Assuming you drop an index by both a name and a key.

Reply:
For more context, DropIndex.command is called when constructing the wire message while executing an operation.